### PR TITLE
[hot-fix] Fixed possible crash when calling metaclient.

### PR DESCRIPTION
**Pull request by monadbobo**
sumarry:

Dropping space while continuously inserting data will trigger this bug.

### CODE REVIEW

File: src/kvstore/PartManager.cpp [hunk 1]
```diff
@@ -13,9 +13,11 @@ PartsMap MemPartManager::parts(const HostAddr& hostAddr) {
...
 
-PartMeta MemPartManager::partMeta(GraphSpaceID spaceId, PartitionID partId) {
+StatusOr<PartMeta> MemPartManager::partMeta(GraphSpaceID spaceId, PartitionID partId) {
     auto it = partsMap_.find(spaceId);
-    CHECK(it != partsMap_.end());
+    if (it == partsMap_.end()) {
+        return Status::Error("Space not found");
+    }
     auto partIt = it->second.find(partId);
     CHECK(partIt != it->second.end());
```
[the remark this answers is not in the document]
Comment (monadbobo) [reply src/kvstore/PartManager.cpp]: Yes, I will fixed this.

File: src/meta/client/MetaClient.cpp [hunk 2]
```diff
@@ -825,13 +825,17 @@ PartsMap MetaClient::getPartsMapFromCache(const HostAddr& host) {
 }
 
 
-PartMeta MetaClient::getPartMetaFromCache(GraphSpaceID spaceId, PartitionID partId) {
+StatusOr<PartMeta> MetaClient::getPartMetaFromCache(GraphSpaceID spaceId, PartitionID partId) {
     folly::RWSpinLock::ReadHolder holder(localCacheLock_);
     auto it = localCache_.find(spaceId);
-    CHECK(it != localCache_.end());
+    if (it == localCache_.end()) {
+        return Status::Error("Space not found");
```
Comment (laura-ding) [inline src/meta/client/MetaClient.cpp]: error msg can include the spaceId and partId

File: src/meta/client/MetaClient.cpp [hunk 3]
```diff
@@ -825,13 +825,17 @@ PartsMap MetaClient::getPartsMapFromCache(const HostAddr& host) {
...
+    }
     auto& cache = it->second;
     auto partAllocIter = cache->partsAlloc_.find(partId);
-    CHECK(partAllocIter != cache->partsAlloc_.end());
+    if (partAllocIter == cache->partsAlloc_.end()) {
+        return Status::Error("Part not found, id was %d", partId);
```
Comment (laura-ding) [inline src/meta/client/MetaClient.cpp]: ditto

File: src/meta/client/MetaClient.cpp [hunk 4]
```diff
@@ -872,15 +876,15 @@ bool MetaClient::checkSpaceExistInCache(const HostAddr& host,
...
 }
 
-
-int32_t MetaClient::partsNum(GraphSpaceID spaceId) {
+StatusOr<int32_t> MetaClient::partsNum(GraphSpaceID spaceId) {
     folly::RWSpinLock::ReadHolder holder(localCacheLock_);
     auto it = localCache_.find(spaceId);
-    CHECK(it != localCache_.end());
+    if (it == localCache_.end()) {
+        return Status::Error("Space not found");
```
Comment (laura-ding) [inline src/meta/client/MetaClient.cpp]: ditto

File: src/storage/client/StorageClient.cpp [hunk 5]
```diff
@@ -398,9 +456,13 @@ folly::Future<StatusOr<cpp2::GetUUIDResp>> StorageClient::getUUID(
     });
 }
 
+StatusOr<PartitionID> StorageClient::partId(GraphSpaceID spaceId, int64_t id) const {
+    auto status = partsNum(spaceId);
+    if (!status.ok()) {
+        return Status::Error("Space not found");
```
Comment (laura-ding) [inline src/storage/client/StorageClient.cpp]: ditto

File: src/kvstore/PartManager.cpp [hunk 6]
```diff
@@ -13,11 +13,15 @@ PartsMap MemPartManager::parts(const HostAddr& hostAddr) {
     return partsMap_;
 }
 
-PartMeta MemPartManager::partMeta(GraphSpaceID spaceId, PartitionID partId) {
+StatusOr<PartMeta> MemPartManager::partMeta(GraphSpaceID spaceId, PartitionID partId) {
     auto it = partsMap_.find(spaceId);
-    CHECK(it != partsMap_.end());
+    if (it == partsMap_.end()) {
+        return Status::Error("Space not found");
```
Comment (laura-ding) [inline src/kvstore/PartManager.cpp]: error msg can include the spaceId and partId

File: src/kvstore/PartManager.cpp [hunk 7]
```diff
@@ -13,11 +13,15 @@ PartsMap MemPartManager::parts(const HostAddr& hostAddr) {
...
+    }
     auto partIt = it->second.find(partId);
-    CHECK(partIt != it->second.end());
+    if (partIt == it->second.end()) {
+        return Status::Error("Part not found in MemPartManager, id was %d", partId);
```
Comment (laura-ding) [inline src/kvstore/PartManager.cpp]: ditto

File: src/meta/client/MetaClient.cpp (hunk 2 above)
Comment (monadbobo) [reply src/meta/client/MetaClient.cpp]: OK，I will fixed This.

File: src/kvstore/PartManager.cpp [hunk 8]
```diff
@@ -16,11 +16,12 @@ PartsMap MemPartManager::parts(const HostAddr& hostAddr) {
...
     auto it = partsMap_.find(spaceId);
     if (it == partsMap_.end()) {
-        return Status::Error("Space not found");
+        return Status::Error("Space not found, spaceid: %d", spaceId);
     }
     auto partIt = it->second.find(partId);
     if (partIt == it->second.end()) {
-        return Status::Error("Part not found in MemPartManager, id was %d", partId);
+        return Status::Error(
+            "Part not found in MemPartManager, spaceid: %d, partId: ", spaceId, partId);
```
Comment (laura-ding) [inline src/kvstore/PartManager.cpp]: missing `%d` after partId:

**Comment by dutor** [comment]
http://ci.vesoft.com/job/Nebula-Graph-UnitTest/2980/console

**Pull request merged**